controllers/serviceclaim_controller.go

```diff
@@ -183,28 +183,70 @@ func (r *ServiceClaimReconciler) processServiceClaim(
 		StringData: map[string]string{},
 	}
 
+	var cel primazaiov1alpha1.ClusterEnvironmentList
+	if err := r.List(ctx, &cel); err != nil {
+		l.Info("error fetching ClusterEnvironmentList", "error", err)
+		return client.IgnoreNotFound(err)
+	}
+
```

Comment:
That is no more needed here, we can move it in `func (r *ServiceClaimReconciler) pushToClusterEnvironments`.
In this function we can also look if the `sclaim.ApplicationClusterContext` is defined or not. If it is we do not need to get all the ClusterEnvironment. We just need the one that we are referencing and we should manage the scenario in which it is not present. In this case a condition should be added.

Reply:
Done